km61/Raievskyi_Yevhenii/laboratory1.sql

```diff
@@ -1,2 +1,127 @@
--- LABORATORY WORK 1
--- BY Raievskyi_Yevhenii
+/*---------------------------------------------------------------------------
+1.Створити схему даних з назвою – прізвище студента, у якій користувач може: 
+створювати таблиці
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+
+CREATE USER raevskiy IDENTIFIED BY 1344
+DEFAULT TABLESPACE "USER"
+TEMPORARY TABLESPACE "TEMP";
+GRANT 'CONNECT' TO raevskiy;
+GRANT CREATE ANY TABLE TO raevskiy;
+
+
+/*---------------------------------------------------------------------------
+2. Створити таблиці, у яких визначити поля та типи. Головні та зовнішні ключі 
+створювати окремо від таблиць використовуючи команди ALTER TABLE. 
+Студент має мобільний номер іноземного оператора.
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+
+CREATE TABLE students
+(
+    stud_name VARCHAR(50)
+);
+
+ALTER TABLE students
+    ADD CONSTRAINT stud_pk PRIMARY KEY (stud_name);
+
+CREATE TABLE numbers
+(
+    phone_number NUMBER(15, 0),
+    phone_country VARCHAR(50)
+);
+
+ALTER TABLE numbers
+    ADD CONSTRAINT numbers_pk PRIMARY KEY (phone_number);
+    
+CREATE TABLE students_numbers
+(
+    stud_name_fk VARCHAR(50),
+    phone_number_fk NUMBER(15, 0)
+);
+
+ALTER TABLE students_numbers
+    ADD CONSTRAINT students_numbers_pk PRIMARY KEY (stud_name_fk, phone_number_fk);
+    
+ALTER TABLE students_numbers
+    ADD CONSTRAINT stud_name_fk FOREIGN KEY (stud_name_fk) REFERENCES students (stud_name);
+    
+ALTER TABLE students_numbers
+    ADD CONSTRAINT phone_number_fk FOREIGN KEY (phone_number_fk) REFERENCES numbers (phone_number);
+  
```

Comment:
_additional attributes(##) to 2 task and regular expressions
 
	students(
stud_name varchar (50) like (^[A-Z][a-z]{2-8}$)
(##)stud_surname varchar(50) like (^[A-Z][a-z]{2-8}$)
(##)stud_birthday number(2) like (^[0-9]{4}-(0[1-9]|1[012])-(0[1-9]|1[0-9]|2[0-9]|3[01]) $)
(##)stud_gender char(1) like (^[MF]$)
	
	students_numbers(
stud_name_fk VARCHAR(50) (^[A-Z][a-z]{2-8}$),
phone_number_fk NUMBER(15, 0) (^[1-7][0-9]{11}$)
(##)series_numb NUMBER(15) (^[0-9]{15}$)
(##)sim_password varchar(50) (^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?!.*\s).*$)

	numbers(
phone_number NUMBER(15, 0) like (^[1-7][0-9]{11}$)
phone_country VARCHAR(50) like (^[A-Z][A-Za-z-]{12}$)
(##)phonename_operat VARCHAR(50) like (^[A-Z][A-Za-z0-9-]{12}$)
(##)phone_payment NUMBER(2) like (^[0-9][0-9]?[,0-9][0-9])?$)